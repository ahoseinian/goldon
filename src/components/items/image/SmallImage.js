import React from 'react'
const SmallImage = ({id}) => <img className="img-thumbnail" src={'/images/50/' + id + '.jpg'} width="60" alt={id}/>
export default SmallImage
