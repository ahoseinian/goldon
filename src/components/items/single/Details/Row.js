import React from 'react';

const SingleRow = ({name, value, measure}) => (
  <tr>
    <th>{name}</th>
    <td>{value} {measure}</td>
  </tr>
);

SingleRow.propTypes = {
  name: React.PropTypes.string.isRequired,
  value: React.PropTypes.oneOfType([
    React.PropTypes.number,
    React.PropTypes.string
  ]),
  measure: React.PropTypes.string
};

export default SingleRow;
