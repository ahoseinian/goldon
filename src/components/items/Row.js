import React from 'react'
import SmallImage from './image/SmallImage'
import Price from './Price'
import styled from 'styled-components'
import {color} from '../theme'
import Icon from '../common/Icon'

const Td = styled.td `
  vertical-align: middle !important;
`
const LeftBorder = styled.div `
  border-left: 1px solid ${color.light};
  margin-left: 1rem;
  padding-left: 1rem;
`

const Row = ({images, name, quantity, price, removeFromCart}) => (
  <tr>
    <Td>
      { images &&
        <SmallImage id={images[0]}/>
      }
    </Td>
    <Td>{name}</Td>
    <Td>
      <div className="d-flex">
        <LeftBorder>
          {quantity}
        </LeftBorder>
        <Price value={price}/>
      </div>
      <Price value={price * quantity}/>
    </Td>
    <Td className="text-center remove-from-cart" onClick={() => removeFromCart()}>
      <Icon name="times" />
    </Td>
  </tr>
)

Row.propTypes = {
  images: React.PropTypes.array,
  price: React.PropTypes.number.isRequired,
  quantity: React.PropTypes.number.isRequired,
  removeFromCart: React.PropTypes.func
}

export default Row
