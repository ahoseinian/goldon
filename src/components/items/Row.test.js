import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Row from './Row'
import SmallImage from './image/SmallImage'

const item = {
  price: 235,
  quantity: 2
}

describe('Row', () => {
  it('renders', () => {
    const wrapper = shallow(<Row {...item}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('does not render SmallImage when there is no image', () => {
    const wrapper = shallow(<Row {...item}/>)
    expect(wrapper.find(SmallImage).length).toEqual(0)
  })

  it('renders SmallImage when there is image', () => {
    const itemWithImages = {...item, images: ['afts']}
    const wrapper = shallow(<Row {...itemWithImages}/>)
    expect(wrapper.find(SmallImage).length).toEqual(1)
  })

  it('calls removeFromCart() on remove-from-cart click', () => {
    const removeFn = jest.fn()
    const itemWithRemoveFromCartProp = {...item, removeFromCart: removeFn}
    const wrapper = shallow(<Row {...itemWithRemoveFromCartProp}/>)
    const removeButton = wrapper.find('.remove-from-cart')
    removeButton.simulate('click')
    expect(removeFn.mock.calls.length).toEqual(1)

  })
})
