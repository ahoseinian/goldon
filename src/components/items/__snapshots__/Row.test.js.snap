exports[`Row renders 1`] = `
<tr>
  <styled.td />
  <styled.td />
  <styled.td>
    <div
      className="d-flex">
      <styled.div>
        2
      </styled.div>
      <Price
        value={235} />
    </div>
    <Price
      value={470} />
  </styled.td>
  <styled.td
    className="text-center remove-from-cart"
    onClick={[Function]}>
    <Icon
      name="times" />
  </styled.td>
</tr>
`;
